polus-generalized-linear-model-plugin/src/main.R

```diff
@@ -0,0 +1,226 @@
+      }
+    }
+    
+    #Convert to ffdf object
+    datasub_ff = as.ffdf(datasub)
+    
+    #Chunk data
+    chunk_data <-make.data(formula(paste(predictcolumn,paste(resp_var,collapse= "+"),sep="~")), datasub_ff, chunksize=chunk)
+    
+    #Model data based on the options selected
+    #Get only main effects of the variables
+    if (glmmethod == 'PrimaryFactors') {
+      if((modeltype == 'gaussian') || (modeltype == 'Gamma') || (modeltype == 'binomial') ||  (modeltype == 'quasibinomial') || (modeltype == 'poisson') || (modeltype == 'quasipoisson') || (modeltype == 'quasi')) {
+        test_glm <- bigglm(formula(paste(predictcolumn,paste(resp_var,collapse= "+"),sep="~")), data = chunk_data, family = eval(parse(text=paste(modeltype,"()", sep = ""))), chunksize = chunk)
+      }
+      else if (modeltype == 'NegativeBinomial') {
+        #Based on glm.nb from MASS package, if theta is not given then poisson GLM is calculated and hence, have considered family as poisson
+        test_glm <- bigglm(formula(paste(predictcolumn,paste(resp_var,collapse= "+"),sep="~")), data = chunk_data, family = poisson(),chunksize=chunk)
+      }
+      else if (modeltype == 'multinomial') {
+        test_glm <- multinom(formula(paste(paste("as.factor(",predictcolumn,")"),paste(resp_var,collapse= "+"),sep="~")), data = datasub)
+        }
+    }
+    #Get interaction values
+    else if (glmmethod == 'Interaction') {
+      datasub_pred <- datasub[ , !(names(datasub) %in% predictcolumn)]
+      #Get correlation between variables
+      tmp <- cor(datasub_pred)
+      tmp[upper.tri(tmp)] <- 0
+      diag(tmp) <- 0
+      
+      #Remove variables with no interaction
+      data_no_int <- which(tmp >= 0.1 | tmp < -0.1, arr.ind = TRUE)
+      data_frame<-data.frame(row = rownames(data_no_int), col = colnames(tmp)[data_no_int[, "col"]],
+                      value = tmp[tmp >= 0.1 | tmp < -0.1])
+      colnames(data_frame)<- c("variable1","variable2","coef")
+      
+      #Interaction variables
+      data_frame$variableint <- paste(data_frame$variable1, data_frame$variable2, sep="*")
+      data_list <- as.character(data_frame$variableint)
+      
+      if((modeltype == 'gaussian') || (modeltype == 'Gamma') || (modeltype == 'binomial') ||  (modeltype == 'quasibinomial') || (modeltype == 'poisson') || (modeltype == 'quasipoisson') || (modeltype == 'quasi')) {
+         test_glm <- bigglm(formula(paste(predictcolumn,paste(data_list,collapse= "+"),sep="~")), data = chunk_data, family = eval(parse(text=paste(modeltype,"()", sep = ""))), chunksize = chunk)
+      }
+      else if (modeltype == 'NegativeBinomial') {
+        test_glm <- bigglm(formula(paste(predictcolumn,paste(data_list,collapse= "+"),sep="~")), data = chunk_data, family = poisson(), chunksize = chunk)
+      }
+      else if (modeltype == 'multinomial') {
+        test_glm <- multinom(formula(paste(predictcolumn,paste(data_list,collapse= "+"),sep="~")), data = datasub, MaxNWts = 100000)
+      }
```

Comment:
This appears to be exactly the same code as was written above. Please eliminate the redundancy by putting this code all in one place.